### PR TITLE
graphics_draw_bitmap_in_rect

### PR DESCRIPTION
**Pull request by Helco**
The previous implementation had some problems (e.g. tiling pebble-dev/RebbleOS#58, compositing modes), so I rewrote it and broke it up into smaller pieces. Also based on my tests/my understanding of the docs these are the modes PebbleOS supports: 

### Black/White
|        | Assign | Invert | And | Or | Clear | Set |
|:----:|:--------:|:------:|:-----:|:---:|:-------:|:---:|
|1Bit |    ✓       |   ✓    |    ✓  |  ✓  |     ✓     |  ✓  |
|1BitPal |  ✓ | X | X | X | X | alpha test |
|2BitPal |  ✓ | X | X | X | X | alpha test |
|4BitPal |  X | X | X | X | X | X |
|8Bit |  X | X | X | X | X | X |

### Color
|        | Assign | Invert | And | Or | Clear | Set |
|:----:|:--------:|:------:|:-----:|:---:|:-------:|:---:|
|1Bit |    ✓       |   undef |    undef|  undef|     undef     |  ✓  |
|1BitPal |  ✓ | X | X | X | X | color blending |
|2BitPal |  ✓ | X | X | X | X | color blending |
|4BitPal |  ✓ | X | X | X | X | color blending |
|8Bit |  ✓ | X | X | X | X | color blending |

Thus these are the modes I implemented for now ( + 4BitPal->BW). We should discuss if we want to support them (maybe even with some fancy dithering?) or keep as compatible as it gets to the old API.
~Also no tests on ARM yet, so this should be regarded as WIP for now~ Works on RebbleOS!

### CODE REVIEW

**Comment by jwise** [comment]
I am ok with having a superset of original Pebble functionality, though we should also balance functionality with code size on flash-constrained devices (like tintin). I'll try to have a look at the code later...

File: contrib/pebble_env/pebble.h [hunk 1]
```diff
@@ -12,6 +12,11 @@
 #include <string.h>
 #include <inttypes.h>
 
+#define NGFX_PREFERRED_free free
+#define NGFX_PREFERRED_calloc calloc
+#define NGFX_PREFERRED_malloc malloc
+#define NGFX_PREFERRED_resource_load resource_load
```
Comment (ginge) [inline contrib/pebble_env/pebble.h:18]: Not sure if I missed it, but I don't see this being used anywhere, but is required in a macro later?

**Review by ginge — COMMENTED** [review]
Looking good. Tests all check out. I'm going to upload to a RebbleOS device next

File: contrib/pebble_env/pebble.h (hunk 1 above)
Comment (Helco) [reply contrib/pebble_env/pebble.h:18]: The macros in this file are used only for the PC testrunner, RebbleOS has to provide its own ones.
The change is meant to move the selection of memory functions out of ngfx.

**Comment by ginge** [comment]
I'm happy to sign this off. The code is clean, I can't break it on device (tintin and snowy tested). I see no reason why this very fine pull request should not be accepted

**Comment by jwise** [comment]
Signed off by me.  Very thorough, and as always, beautiful code.  Feel free to merge, @Helco.

**Review by jwise — APPROVED** [review]

**Pull request merged**